contributors/design-proposals/cluster-info-configmap.md

```diff
@@ -0,0 +1,25 @@
+# Persist cluster-wide data in ConfigMap
+
+## Abstract
+
+We want to allow cluster-wide data to be persisted for general purposes, currently for us (service catalog) is to focus on the ability to identify different clusters that access our service broker with a 'cluster-id'.
+
+## Motivation
+
+Currently there is no way for service catalog api server to identify the different clusters that access our service brokers. The goal of this design is to have a unify storage for cluster related information. The information should be retained and survive between component failures. 
+
+## Detailed design
+
+A configmap `cluster-info` is created in the `default` namespace at the starup of the controller. The initial content of the map will be as the following:
```

Comment:
Actually, updating the ConfigMap's clusterID will work. Is there a way to make it so its read-only once its set (if that's what we want)? I can envision cases where upon restoring a system an admin may need to reset the cluster's ID to a previous value.